lib/clientv2/resources.go

```diff
@@ -271,11 +295,77 @@ func (c *resources) handleNamespace(ns, kind string, in resource) error {
 	return nil
 }
 
+type watcher struct {
+	backend bapi.WatchInterface
+	context context.Context
+	cancel  context.CancelFunc
+	results chan watch.Event
+	client  *resources
+}
+
+func (w *watcher) Stop() {
+	w.cancel()
+}
+
+func (w *watcher) ResultChan() <-chan watch.Event {
+	return w.results
+}
+
+func (w *watcher) run() {
+	log.Info("Main client watcher loop")
+mainloop:
+	for {
+		select {
+		case event := <-w.backend.ResultChan():
+			e := w.processEvent(event)
```

Comment:
could this be a goroutine since we're waiting with a `select{}` later? `e := go w.processEvent(event)`

Reply:
Nope - the events would then potentially arrive out of order I think

The select is the processEvent is blocking on the results channel being backed up, in which case we'd end up spawning potentially a LOT of go routines.

Reply:
ah ok